solana/program_v2/programs/gateway_v2/src/instructions/network/gatekeeper_withdraw.rs

```diff
@@ -1,18 +1,19 @@
 use crate::constants::GATEKEEPER_SEED;
-use crate::state::Gatekeeper;
+use crate::errors::GatekeeperErrors;
+use crate::state::{Gatekeeper, GatekeeperKeyFlags};
 use anchor_lang::prelude::*;
-use anchor_spl::token::{Mint, Token, TokenAccount, Transfer};
+use anchor_spl::token::{Token, TokenAccount, Transfer};
 
 // Will withdraw funds from the gatekeeper
 pub fn gatekeeper_withdraw(ctx: Context<GatekeeperWithdrawAccount>, mut amount: u64) -> Result<()> {
     // Amount 0 means withdraw all
     if amount == 0 {
         amount = ctx.accounts.gatekeeper_token_account.amount;
     }
-    let gatekeeper = &mut ctx.accounts.gatekeeper;
-    let spl_token_program = &mut ctx.accounts.spl_token_program;
-    let receiver_token_account = &mut ctx.accounts.receiver_token_account;
-    let gatekeeper_token_account = &mut ctx.accounts.gatekeeper_token_account;
+    let gatekeeper = &ctx.accounts.gatekeeper;
+    let spl_token_program = &ctx.accounts.spl_token_program;
+    let receiver_token_account = &ctx.accounts.receiver_token_account;
+    let gatekeeper_token_account = &ctx.accounts.gatekeeper_token_account;
```

Comment:
This and receiver_token_account are not set to be mutable here, but are set as mutable in the accounts list. Why are they mutable one place and not at another? Could be I am missing something

Reply:
because this program does not mut the account, but the CPI transfer does.

Reply:
Gotcha, thanks!